pkg/scheduler/plugins/deviceshare/device_cache.go

```diff
@@ -382,15 +382,20 @@ func newNodeDeviceCache() *nodeDeviceCache {
 	}
 }
 
-func (n *nodeDeviceCache) getNodeDevice(nodeName string) *nodeDevice {
-	n.lock.RLock()
-	defer n.lock.RUnlock()
+func (n *nodeDeviceCache) getNodeDevice(nodeName string, needInit bool) *nodeDevice {
+	n.lock.Lock()
```

Comment:
Please change the lock type to sync.Mutex

Reply:
changed to sync.Mutex